### DISCUSSION TITLE
Cannot start iPXE using UEFI on VMWare ESXi 6.7, legacy works

### DISCUSSION THREAD

**Discussion by synedraIT** [ipxe/ipxe#442] [Q&A] [answered]
Hello!
I wanted to use iPXE to boot up various live CDs and installation media in our environment.
PXE with Legacy boot already works wonderfully here, iPXE gets started and can use it's config.

However, on VMWare, if I use a VM with UEFI to boot (which most of our servers are), I get no prompt. I already followed the tutorial here: https://ipxe.org/howto/chainloading
The DHCP-Server already sends the correct file out and the PXE ROM by vmware also tries to load it - I checked this with wireshark:
![image](https://user-images.githubusercontent.com/66822927/129326606-24eb023e-8c5e-4c2d-8376-51f9ebc2abd4.png)
...but it somehow doesn't work, after briefly flashing this:
![image](https://user-images.githubusercontent.com/66822927/129327291-4f9eb46f-1c24-4ef7-907d-ec30cdfbe719.png)
it goes back to BIOS:
![image](https://user-images.githubusercontent.com/66822927/129327391-86b37e17-8d60-4aaf-94bb-d382cf2ea7bb.png)

I can't say whether or not it works on non-vmware, as I am currently working remote.

edit:
These are the VM settings:
![image](https://user-images.githubusercontent.com/66822927/129327857-6476bbc2-48f1-47fc-9a70-18231c43467f.png)

**Comment by NiKiZe** [comment, marked as answer]
this at first looks like a secure boot issue, but that is of according to your settings.
Please try with snponly.efi instead (grab it from boot.ipxe.org just as a test)

**Reply by synedraIT** [reply]
works!
Thank you very much! I'll try compiling it on Monday and let you know how it goes!

**Reply by NiKiZe** [reply]
Ok, that was not a suggested solution tho, just collecting data, But great that it worked.
You might want to try with ipxe.efi from the same place and see if that works. If it does then it might be an issue with your compiled version.
(this is to narrow down where the actual issue is)

**Reply by synedraIT** [reply]
Oh, sorry, I thought that was the solution.
I already tried the ipxe.efi from there, it did not work - same as with the compiled version.

**Reply by NiKiZe** [reply]
So ipxe.efi hangs even before "iPXE initializing devices ..." line?
Could you try with different nics? And maybe even give us output from `lspci -nn` when it hangs?

**Reply by synedraIT** [reply]
Okay, did some testing (involing lots of sha256sum), it appears that I mixed some files up.

Both versions from boot.ipxe.org seem to work, meaning I just mixed up a self compiled version of ipxe in there.
Thank you very much for your patience and sorry for the trouble!
